### PR TITLE
Add Variety to Az.Predictor

### PR DESCRIPTION
**Pull request by yemohyle**
## Description

Fixed issue of duplicate threshold consistency caused by two Query function calls as well as duplicate tolerance when list of novel commands is insufficiently small

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by kceiw — CHANGES_REQUESTED** [review]

File: tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictorService.cs [hunk 1]
```diff
@@ -125,14 +125,17 @@ protected virtual void Dispose(bool disposing)
...
         /// Queries the Predictor with the user input if predictions are available, otherwise uses commands
         /// </remarks>
-        public IEnumerable<ValueTuple<string, string, PredictionSource>> GetSuggestion(Ast input, int suggestionCount, CancellationToken cancellationToken)
+        public IEnumerable<ValueTuple<string, string, PredictionSource>> GetSuggestion(Ast input, int suggestionCount, int maxAllowedCommandDuplicate, CancellationToken cancellationToken)
         {
             var commandSuggestions = this._commandSuggestions;
             var command = this._commandForPrediction;
 
             IList<ValueTuple<string, string, PredictionSource>> results = new List<ValueTuple<string, string, PredictionSource>>();
+            var presentCommands = new System.Collections.Generic.Dictionary<string, int>(); // Added
```
[the remark this answers is not in the document]
Comment (yemohyle) [reply tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictorService.cs]: Removed

File: tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictorSettings.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
-  "serviceUri":  "https://app.aladdin.microsoft.com/api/v1",
-  "suggestionCount": 7
+  "serviceUri": "https://app.aladdin.microsoft.com/api/v1",
+  "suggestionCount": 7,
+  "maxAllowedCommandDuplicate": 1
```
[the remark this answers is not in the document]
Comment (yemohyle) [reply tools/Az.Tools.Predictor/Az.Tools.Predictor/AzPredictorSettings.json]: Moved

File: tools/Az.Tools.Predictor/Az.Tools.Predictor/Predictor.cs [hunk 3]
```diff
@@ -57,11 +57,13 @@ public Predictor(IList<string> modelPredictions, ParameterValuePredictor paramet
...
         /// </summary>
         /// <param name="input">PowerShell AST input of the user, generated by PSReadLine</param>
+        /// <param name="presentCommands">Commands already present.</param>
         /// <param name="suggestionCount">The number of suggestion to return.</param>
+        /// <param name="maxAllowedCommandDuplicate">The maximum amount of the same commnds in the list of predictions.</param>
         /// <param name="cancellationToken">The cancellation token</param>
         /// <returns>The collection of suggestions. The key is the predicted text adjusted based on <paramref name="input"/>. The
         /// value is the original text to create the adjusted text. </returns>
-        public IDictionary<string, string> Query(Ast input, int suggestionCount, CancellationToken cancellationToken)
+        public Tuple<IDictionary<string, string>, Dictionary<string, int>> Query(Ast input, Dictionary<string, int> presentCommands, int suggestionCount, int maxAllowedCommandDuplicate, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (yemohyle) [reply tools/Az.Tools.Predictor/Az.Tools.Predictor/Predictor.cs]: Changed

File: tools/Az.Tools.Predictor/Az.Tools.Predictor/Predictor.cs [hunk 4]
```diff
@@ -102,6 +105,7 @@ public Predictor(IList<string> modelPredictions, ParameterValuePredictor paramet
                 var resultBuilder = new StringBuilder();
                 var usedParams = new HashSet<int>();
                 var sourceBuilder = new StringBuilder();
+                //var presentCommands = new System.Collections.Generic.Dictionary<string, int>(); // Added
```
[the remark this answers is not in the document]
Comment (yemohyle) [reply tools/Az.Tools.Predictor/Az.Tools.Predictor/Predictor.cs]: Removed

**Review by kceiw — DISMISSED** [review]

**Review by kceiw — APPROVED** [review]

**Review by msJinLei — APPROVED** [review]

**Pull request merged**